components/org.wso2.identity.webhook.common.event.handler/src/main/java/org/wso2/identity/webhook/common/event/handler/internal/handler/TokenEventHookHandler.java

```diff
@@ -116,6 +116,12 @@ private void handleEventForProfile(Event event, EventData eventData, EventProfil
                 .map(org.wso2.carbon.identity.webhook.metadata.api.model.Event::getEventUri)
                 .orElse(null);
 
+        if (EventHookHandlerUtils.isSubOrgLevel()) {
+            log.debug("Event trigger for sub organization level. Skipping event handling for token event profile: " +
```

Comment:
add is debug enabled check since eventProfile.getProfile()); is memory bound

Reply:
How come ?

<img width="1348" height="503" alt="image" src="https://github.com/user-attachments/assets/f43fcded-a869-4356-b86f-f337f05d017a" />

Reply:
minor perf improvement so +1